src/praisonai-bot/praisonai_bot/integration/context_files.py

```
@@ -136,6 +150,128 @@ def _add(path: Path) -> None:
+def _fetch_remote_source(url: str) -> Optional[str]:
+    """Fetch a remote instruction source, best-effort and size-bounded.
+
+    Uses the stdlib ``urllib`` (lazily imported) so no heavy dependency is
+    added. A slow, unreachable, or oversized response is handled gracefully:
+    failures return ``None`` (with a warning) so the run continues, and bodies
+    larger than ``_REMOTE_MAX_BYTES`` are truncated with a marker.
+    """
+    try:
+        import urllib.request
+
+        with urllib.request.urlopen(url, timeout=_REMOTE_TIMEOUT) as resp:  # nosec B310
```

Ruff (0.16.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)